client/gutenberg/extensions/presets/jetpack/editor.js

```diff
@@ -5,8 +5,19 @@
  */
 import './shared/public-path';
 import './editor-shared/block-category'; // Register the Jetpack category
-import 'gutenberg/extensions/contact-form/editor';
-import 'gutenberg/extensions/markdown/editor';
-import 'gutenberg/extensions/map/editor';
-import 'gutenberg/extensions/publicize/editor';
-import 'gutenberg/extensions/simple-payments/editor';
+
+// TODO: Generate dyanmically from index.json
+// Appending `Block` to the names to keep `Map` from colliding with JS' Map
+import * as ContactFormBlock from 'gutenberg/extensions/contact-form/editor';
+import * as MarkdownBlock from 'gutenberg/extensions/markdown/editor';
+import * as MapBlock from 'gutenberg/extensions/map/editor';
+import * as PublicizeBlock from 'gutenberg/extensions/publicize/editor';
+import * as SimplePaymentsBlock from 'gutenberg/extensions/simple-payments/editor';
+
+export default {
+	[ ContactFormBlock.name ]: ContactFormBlock.settings,
+	[ MarkdownBlock.name ]: MarkdownBlock.settings,
+	[ MapBlock.name ]: MapBlock.settings,
+	[ PublicizeBlock.name ]: PublicizeBlock.settings,
+	[ SimplePaymentsBlock.name ]: SimplePaymentsBlock.settings,
```

Comment:
We can remove all the blocks from here and only leave the `SimplePaymentsBlock` since that is the only block for now that can be toggled on and off. 

All the other blocks will be registered always when we load the all the editor files.

Reply:
TBH I'd rather leave them in here -- it's one step towards a more declarative way of registering blocks IMO (explicitly exporting relevant symbols here for use by a consumer, rather than just importing them and relying on their implicit module-scope invocation of `registerJetpackBlock` or `registerJetpackPlugin`).

Also, I think it helps to keep the multiple lists of blocks we currently maintain aligned so we won't be scratching our heads why one list contains a different set of blocks than another when consolidating them.

Reply:
I second @ockham — for better consistency and it's a good step towards having one source of truth for the blocks https://github.com/Automattic/wp-calypso/issues/28942 (only json files, instead of json and editor.js)